core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -270,6 +270,11 @@ void contractNodes() {
                         + ", algo:" + prepareAlgo.getMemoryUsageAsString()
                         + ", " + Helper.getMemInfo());
 
+                if(Thread.interrupted()){
```

Comment:
Here we should use Thread.isInterrupted instead. Also we should check it more frequently otherwise it will be check just ~4 times per contraction. I expect no slow down here :)